### PR TITLE
[Merged by Bors] - Fix resource limits test

### PR DESCRIPTION
**Pull request by adwk67**
# Description

Assert resource limits using Pod definitions

## Review Checklist

- [x] Code contains useful comments
- [x] CRD change approved (or not applicable)
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: tests/templates/kuttl/resources/10-assert.yaml [hunk 1]
```diff
@@ -3,10 +3,82 @@ apiVersion: kuttl.dev/v1beta1
...
+          memory: 1Gi
+        requests:
+          cpu: 100m
+          memory: 1Gi
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    spark-app-name: sparkals
```
Comment (razvan) [inline tests/templates/kuttl/resources/10-assert.yaml:53]: Is this correct ? Is spark overriding this label ?

**Review by razvan — COMMENTED** [review]
lgtm. But I would start a CI run before merging.

**Comment by adwk67** [comment]
> lgtm. But I would start a CI run before merging.

Jenkins: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/spark-k8s-operator-it-custom/37/

File: tests/templates/kuttl/resources/10-assert.yaml (hunk 1 above)
Comment (adwk67) [reply tests/templates/kuttl/resources/10-assert.yaml:53]: I don't think we set that (at least I couldn't find any reference to that in the code) - that seems to come from the main application class.

**Review by maltesander — APPROVED** [review]
LGTM.

File: tests/templates/kuttl/resources/10-assert.yaml (hunk 1 above)
Comment (adwk67) [reply tests/templates/kuttl/resources/10-assert.yaml]: I think you might be right: I had some issues with slightly different labels being assigned by KinD / managed k8s. I'll look into separately.

**Comment by adwk67** [comment]
bors merge

**Pull request closed without merging**